internal/transport/http2_client.go

```diff
@@ -505,9 +507,18 @@ func (t *http2Client) createHeaderFields(ctx context.Context, callHdr *CallHdr)
 		headerFields = append(headerFields, hpack.HeaderField{Name: "grpc-previous-rpc-attempts", Value: strconv.Itoa(callHdr.PreviousAttempts)})
 	}
 
+	advertisedCompressors := t.advertisedCompressors
 	if callHdr.SendCompress != "" {
 		headerFields = append(headerFields, hpack.HeaderField{Name: "grpc-encoding", Value: callHdr.SendCompress})
-		headerFields = append(headerFields, hpack.HeaderField{Name: "grpc-accept-encoding", Value: callHdr.SendCompress})
+		// Advertise the outgoing compressor name, when compressor is not registered
+		// via encoding.RegisterCompressor.
+		if !grpcutil.IsCompressorNameRegistered(callHdr.SendCompress) {
+			advertisedCompressors = callHdr.SendCompress
```

Comment:
Assuming it is possible, shouldn't this append instead of replace?

Reply:
I didn't replace to avoid string allocation for every call involving a compressor not set through `encoding.RegisterCompressor`. But in terms of correctness, it is right to replace.